pwndbg/tests/test_misc.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import pytest
+
+from pwndbg.services.misc import pwndbg_list_and_filter_commands
+
+
+def test_show_super_assert():
+    x = [1, 2, 3]
+    assert x == [1, 2, 4]
+
+
+def test_code_that_should_fail():
+    with pytest.raises(ZeroDivisionError):
+        1 / 0
+
+
+def test_pwndbg_list_and_filter_commands_filter():
+    assert pwndbg_list_and_filter_commands('stack') == [
+        ('canary', 'Print out the current stack canary'),
+        ('context', 'Print out the current register, instruction, and stack context.'),
+        ('down', 'Select and print stack frame called by this one.'),
+        ('retaddr', 'Print out the stack addresses that contain return addresses'),
+        ('stack', 'dereferences on stack data with specified count and offset'),
+        ('up', 'Select and print stack frame that called this one.')
+    ]
+
+
+def test_pwndbg_list_and_filter_commands_full_list():
+    assert pwndbg_list_and_filter_commands('') == [
```

Comment:
This looks like a giant PITA to maintain.  I'd recommend picking a single command, and using `.__doc__`  to fetch the expected string.